src/features/dynamic-import.md

```diff
@@ -42,7 +42,7 @@ Here’s how to statically import and use the `./utils.mjs` module:
 ```
 
 :::note
-**Note:** The previous example uses the `.mjs` extension to signal that it’s a module rather than a regular script. On the web, file extensions don’t really matter, as long as the files are served with the correct MIME type (e.g. `text/javascript` for JavaScript files) in the `Content-Type` HTTP header. The `.mjs` extension is [especially useful](https://github.com/nodejs/node-eps/blob/master/002-es-modules.md#32-determining-if-source-is-an-es-module) on other platforms such as Node.js, where there’s no concept of MIME types or other mandatory hooks such as `type="module"` to determine whether something is a module or a regular script. We’re using the same extension here for consistency across platforms and to clearly make the distinction between modules and regular scripts.
```

Comment:
Seems like we should just keep this, and update the link to <https://nodejs.org/api/esm.html#esm_enabling>. We could also point out that `d8` recognizes `.mjs`, in case we want to add more examples than just Node.js.

Reply:
I’ve updated my revision per your note.

The “no concept of MIME types or other mandatory hooks such as `type="module"`” language isn’t quite correct. As you know, Node has `"type": "module"` now, which in fairness didn’t exist back when this sentence was written; but before you argue that the current text is still correct because of the “mandatory” qualifier, keep in mind that HTML’s `type="module"` isn’t mandatory either: a regular script can load a module via `import()`, with no `type="module"` in sight.

But that’s just one clause, and the paragraph is stronger without it. `.mjs`’s utility is that it indicates a file’s module-ness for _all_ non-browser environments and tools, rather than Node.js’ `"type": "module"` or Spidermonkey’s `-m` or the various proprietary configuration options in Babel and Webpack and so on. I think that point comes across more clearly without the digression into `.mjs`’s competitors.

Reply:
> The “no concept of MIME types or other mandatory hooks such as `type="module"`” language isn’t quite correct. As you know, Node has `"type": "module"` now, which in fairness didn’t exist back when this sentence was written; but before you argue that the current text is still correct because of the “mandatory” qualifier, keep in mind that HTML’s `type="module"` isn’t mandatory either: a regular script can load a module via `import()`, with no `type="module"` in sight.

Yeah, but the script doing the dynamic `import()` would be a classic script, not a module. The current phrasing is precise and correct.

Reply:
Allow me to be blunt, if you don’t mind. The original sentence was this:

> The `.mjs` extension is especially useful on other platforms such as Node.js, where there’s no concept of MIME types or other hooks such as `type="module"` to determine whether something is a module or a regular script.

Then `"type": "module"` came to Node, and so that sentence was no longer true. So you [added](https://github.com/v8/v8.dev/commit/e3ba0f22395d4141a6f61ae8aa17f234c29da497#diff-356674ffcf237bcfbc2efdc7a09c7ea1R45) “mandatory” before “hooks.”

As currently written, “mandatory” is a weasel word. It makes the sentence true again (arguably) but it is misleading to the reader in that it implies that `.mjs` is the only way to use ESM in Node. No reasonable reader would read this sentence and assume that there was any way to use ESM in Node other than via `.mjs`. No one would read this and think, “oh, but does Node have any _non_-mandatory hooks?”

If you want to recommend `.mjs`, that’s fine; my revised text makes that case for you, without deceptive language.

Reply:
P.S. to my last comment: I didn’t mean to imply that you were being deceptive in your edit. I only meant that it comes across that way to me, and I’m hardly unbiased. Apologies if I came across otherwise.

We’re debating the wrong thing. The question isn’t whether the current text is wrong or incorrect; the question should be whether the new text is _better._ So let’s compare:

> **Old:** The `.mjs` extension is especially useful on other platforms such as Node.js, where there’s no concept of MIME types or other mandatory hooks such as `type="module"` to determine whether something is a module or a regular script.

> **New:** The `.mjs` extension is especially useful on other platforms such as Node.js and `d8`, where it’s a convenient way to indicate to the runtime that your code is a module.

Both sentences offer a reason why `.mjs` is especially useful. The old sentence’s reason is why `.mjs` is useful _for platforms:_ `.mjs` is useful for such platforms to have a way to be able to recognize modules. The new sentence’s reason is why `.mjs` is useful _for users:_ this is a convenient way to signal to platforms that a file is a module.

As this article is written for users, I think the latter sentence is more appropriate.